lightning/src/onion_message/messenger.rs

```diff
@@ -1267,15 +1272,15 @@ where
 	pub fn handle_onion_message_response<T: OnionMessageContents>(
 		&self, response: ResponseInstruction<T>
 	) -> Result<Option<SendSuccess>, SendError> {
-		let (response, create_reply_path) = match response {
-			ResponseInstruction::WithReplyPath(response) => (response, true),
-			ResponseInstruction::WithoutReplyPath(response) => (response, false),
+		let (response, context, create_reply_path) = match response {
+			ResponseInstruction::WithReplyPath(response, context) => (response, Some(context), true),
+			ResponseInstruction::WithoutReplyPath(response) => (response, None, false),
 			ResponseInstruction::NoResponse => return Ok(None),
 		};
 
 		let message_type = response.message.msg_type();
 		let reply_path = if create_reply_path {
-			match self.create_blinded_path() {
+			match self.create_blinded_path(context.unwrap()) {
```

Comment:
Instead of keeping `create_reply_path`, let's drop it in favor of using `context`. That way we don't need to `unwrap` here, which is best to avoid.

Reply:
Makes sense!
Updated in [c8de6b6](https://github.com/lightningdevkit/rust-lightning/commit/c8de6b610fdde71730c5f536ecd89e9afcf003ea)
Thanks for the pointer!